### ISSUE TITLE
Issues with Bounding Box Coordinates Exceeding Image Dimensions After ONNX Export

### ISSUE THREAD

**Issue by Venki-code29** [Deci-AI/super-gradients#1998]
### 💡 Your Question

Hey guys, I am currently working on an object detection project. The model performs pretty good during training and inference using the .predict method, however, I am encountering issues after exporting the model to ONNX format. Specifically, some of the bounding box coordinates exceed the dimensions of the input image. Here are the details of my workflow and the problems I am facing: 

1) I exported the trained model to ONNX format using the following parameters:
export_result = best_model.export(output="run1.onnx",nms_threshold=0.7, confidence_threshold=0.6)
I am including preprocessing and postprocessing steps in the exported model.

2) I preprocess the image before feeding it into the ONNX model using the following steps:
original_image = cv2.imread(r'XXXX.jpg')
original_height, original_width = original_image.shape[:2]

input_image = cv2.resize(original_image, (640, 640))
input_image = cv2.cvtColor(input_image, cv2.COLOR_BGR2RGB)  # Convert BGR to RGB
input_image = np.transpose(input_image, (2, 0, 1))   # Transpose to (3, 640, 640)
input_image = np.expand_dims(input_image, axis=0)  # Add batch dimension (1, 3, 640, 640)
input_image = input_image.astype(np.uint8)

original_image.shape
(3300, 2550, 3)

input_image.shape
(1, 3, 640, 640)

3) I run the ONNX model using the following code:
import onnxruntime
import numpy as np
session = onnxruntime.InferenceSession("myexport.onnx", providers=["CUDAExecutionProvider", "CPUExecutionProvider"])
inputs = [o.name for o in session.get_inputs()]
outputs = [o.name for o in session.get_outputs()]
predictions = session.run(outputs, {inputs[0]: input_image})

4) I extract the predictions as given in the documentation:
num_detections, pred_boxes, pred_scores, pred_classes = predictions
for image_index in range(num_detections.shape[0]):
    for i in range(num_detections[image_index,0]):
        class_id = pred_classes[image_index, i]
        confidence = pred_scores[image_index, i]
        x_min, y_min, x_max, y_max = pred_boxes[image_index, i]
        print(f"Detected object with class_id={class_id}, confidence={confidence}, x_min={x_min}, y_min={y_min}, x_max={x_max}, y_max={y_max}")

While most bounding boxes are within expected ranges, one of the bounding boxes have y_max values that exceed the image dimensions (640x640). For example:
Detected object with class_id=0, confidence=0.8235642910003662, x_min=393.44537353515625, y_min=61.27360534667969, x_max=514.3463134765625, y_max=654.8359985351562

The y_max value is 654.836, which is beyond the 640-pixel height of the input image. This issue does not occur during inference with the .predict method before exporting to ONNX. I have tried ONNX_flat als

Can you please shed light on why the bounding box coordinates occasionally exceed the image dimensions after ONNX export. Could there be an issue with the post-processing implementation during the export, or is there something I am missing in the inference code?

Any guidance or suggestions to resolve this issue would be greatly appreciated.

Thank you for your assistance!

### Versions

PyTorch version: 2.2.1+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro for Workstations
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22621-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: Quadro P4000
Nvidia driver version: 537.70
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9
CurrentClockSpeed=2095
DeviceID=CPU0
Family=179
L2CacheSize=16384
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2095
Name=Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz
ProcessorType=3
Revision=21764

Architecture=9
CurrentClockSpeed=2095
DeviceID=CPU1
Family=179
L2CacheSize=16384
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2095
Name=Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz
ProcessorType=3
Revision=21764

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] onnx==1.15.0
[pip3] onnx-graphsurgeon==0.3.27
[pip3] onnxruntime==1.15.0
[pip3] onnxsim==0.4.36
[pip3] torch==2.2.1+cu118
[pip3] torchaudio==2.2.1+cu118
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.17.1+cu118
[conda] Could not collect
